design/cli-install-changes.md

```diff
@@ -0,0 +1,156 @@
+The naming and organization of the proposed new CLI commands below have been inspired on the `kubectl` commands, particularly `kubectl set` and `kubectl config`.
+
+#### Grouping commands
+
+Below is the proposed set of new commands to install and configure Velero.
+
+1) `velero init`
+Configures up the namespace, RBAC, deployment, etc., but does not add any external plugins, BSL/VSL definitions. This would be the minimum set of commands to get the Velero server up and running and ready to accept other configurations. Mostly things to be run once at setup time. Could be named something else, like `install`.
+
+These ones might make sense to include under all the other commands
+```
+      --dry-run                                    generate resources, but don't send them to the cluster. Use with -o. Optional.
+      -h, --help                                   help for install
+      --label-columns stringArray                  a comma-separated list of labels to be displayed as columns
+      -o, --output string                          Output display format. For create commands, display the object but do not send it to the server. Valid formats are 'table', 'json', and 'yaml'. 'table' is not valid for the install command.
+      --wait                                       wait for Velero deployment to be ready. Optional.
+      --show-labels                                show labels in the last column
+```
+
+Minimum set for initialization
+```
+      --image string                               image to use for the Velero and restic server pods. Optional. (default "velero/velero:latest")
+      --sa-annotations mapStringString             annotations to add to the Velero ServiceAccount. Add iam.gke.io/gcp-service-account=[GSA_NAME]@[PROJECT_NAME].iam.gserviceaccount.com for workload identity. Optional. Format is key1=value1,key2=value2
+```
+   
+2) `velero set`
+All the other configuration that is not component specific nor necessary for initialization. Mostly things to be updated. Might be grouped under `init`.
+
+```
+      --pod-annotations mapStringString            annotations to add to the Velero and restic pods. Optional. Format is key1=value1,key2=value2
+      --restore-only                               run the server in restore-only mode. Optional.
+      --pod-cpu-limit string                       CPU limit for Velero pod. A value of "0" is treated as unbounded. Optional. (default "1000m")
+      --pod-cpu-request string                     CPU request for Velero pod. A value of "0" is treated as unbounded. Optional. (default "500m")
+      --pod-mem-limit string                       memory limit for Velero pod. A value of "0" is treated as unbounded. Optional. (default "256Mi")
+      --pod-mem-request string                     memory request for Velero pod. A value of "0" is treated as unbounded. Optional. (default "128Mi")
+```
+
+3) `velero config`
```

Comment:
some of this overlaps with the existing `velero backup-location get/create` and `velero snapshot-location get/create` commands - thoughts on how they relate?

Reply:
Good observation. I would deprecate those and have all of those operations only under `velero config`.

Reply:
To keep in line with the `velero [resource] [operation] [flags]` format, I had to remove the `velero config` and so we are keeping the original `velero backup-location/snapshot-location` with some minor flag modifications/additions.